quarkus-integration/quarkus/deployment/src/main/java/ai/timefold/solver/quarkus/deployment/TimefoldProcessor.java

```diff
@@ -148,20 +152,48 @@ DetermineIfNativeBuildItem ifNativeBuild() {
     }
 
     @BuildStep(onlyIf = IsDevelopment.class)
-    public DevConsoleRuntimeTemplateInfoBuildItem getSolverConfig(SolverConfigBuildItem solverConfigBuildItem,
-            CurateOutcomeBuildItem curateOutcomeBuildItem) {
+    @Record(STATIC_INIT)
+    public CardPageBuildItem registerDevUICard(
+            TimefoldDevUIRecorder devUIRecorder,
+            SolverConfigBuildItem solverConfigBuildItem,
+            BuildProducer<SyntheticBeanBuildItem> syntheticBeans) {
+        CardPageBuildItem cardPageBuildItem = new CardPageBuildItem();
         SolverConfig solverConfig = solverConfigBuildItem.getSolverConfig();
         if (solverConfig != null) {
             StringWriter effectiveSolverConfigWriter = new StringWriter();
             SolverConfigIO solverConfigIO = new SolverConfigIO();
             solverConfigIO.write(solverConfig, effectiveSolverConfigWriter);
-            return new DevConsoleRuntimeTemplateInfoBuildItem("solverConfigProperties",
-                    new TimefoldDevUIPropertiesSupplier(effectiveSolverConfigWriter.toString()), this.getClass(),
-                    curateOutcomeBuildItem);
+            syntheticBeans.produce(SyntheticBeanBuildItem.configure(SolverConfigText.class)
+                    .scope(ApplicationScoped.class)
+                    .supplier(devUIRecorder.solverConfigTextSupplier(effectiveSolverConfigWriter.toString()))
+                    .done());
         } else {
-            return new DevConsoleRuntimeTemplateInfoBuildItem("solverConfigProperties",
-                    new TimefoldDevUIPropertiesSupplier(), this.getClass(), curateOutcomeBuildItem);
+            syntheticBeans.produce(SyntheticBeanBuildItem.configure(SolverConfigText.class)
+                    .scope(ApplicationScoped.class)
+                    .supplier(devUIRecorder.solverConfigTextSupplier(""))
+                    .done());
         }
+        cardPageBuildItem.addPage(Page.webComponentPageBuilder()
+                .title("Configuration")
+                .icon("font-awesome-solid:wrench")
+                .componentLink("config-component.js"));
+
+        cardPageBuildItem.addPage(Page.webComponentPageBuilder()
+                .title("Model")
+                .icon("font-awesome-solid:wrench")
+                .componentLink("model-component.js"));
+
+        cardPageBuildItem.addPage(Page.webComponentPageBuilder()
+                .title("Constraints")
+                .icon("font-awesome-solid:wrench")
+                .componentLink("constraints-component.js"));
+
+        return cardPageBuildItem;
+    }
+
+    @BuildStep(onlyIf = IsDevelopment.class)
+    public JsonRPCProvidersBuildItem registerRPCService() {
+        return new JsonRPCProvidersBuildItem("Timefold", TimefoldDevUIPropertiesRPCService.class);
```

Comment:
Shall we make it "Timefold Solver"? That way, it'd be consistent with what the project is actually called.